pykickstart/commands/snapshot.py

```diff
@@ -68,28 +85,60 @@ def __str__(self):
         return retval
 
     def _getParser(self):
+        def when_cb(option, opt_str, value, parser):
+            if value.lower() in self.whenMap:
+                parser.values.ensure_value(option.dest,
+                                           self.whenMap[value.lower()])
```

Comment:
Truth is I didn't test that, I copied it from `Autopart` class `type` parameter. And it is there for some time, so I *guess* it should work properly.